administrator/components/com_associations/models/associations.php

```diff
@@ -147,31 +147,21 @@ protected function getListQuery()
 		$columns  = $db->getTableColumns($table);
 		$title    = isset($columns['title']) ? 'a.title' : 'a.name';
 		$ordering = isset($columns['lft']) ? 'a.lft' : 'a.ordering';
+		$select   = array();
 
-		if ($table == '#__menu' || $table == '#__categories')
+		$select[] = $db->quoteName('a.id');
```

Comment:
an array is not needed
you can put 
$query->select(...)
and then after continue the select with
$query->select(...)
it will have the behaviour you intend.

but the principle is a nice optimization